src/moonlink/src/storage/mooncake_table/transaction_stream.rs

```diff
@@ -343,36 +342,36 @@ impl MooncakeTable {
         Ok(disk_slice)
     }
 
-    /// Flushes a disk slice for streaming transaction.
-    /// Increments the pending flush count for this transaction.
-    pub async fn flush_stream_disk_slice(
-        &mut self,
-        xact_id: u32,
-        disk_slice: &mut DiskSliceWriter,
-    ) -> Result<()> {
-        let stream_state = self
-            .transaction_stream_states
-            .get_mut(&xact_id)
-            .expect("Stream state not found for xact_id: {xact_id}");
-        stream_state.pending_flush_count += 1;
-        disk_slice.write().await?;
-        Ok(())
-    }
-
     /// Applies the result of a streaming flush to the stream state.
     /// Decrements the pending flush count for this transaction.
     /// Handles commit and abort cleanup.
     /// Removes in memory indices and record batches from the stream state.
     pub fn apply_stream_flush_result(&mut self, xact_id: u32, mut disk_slice: DiskSliceWriter) {
+        if let Some(lsn) = disk_slice.lsn() {
```

Comment:
(definitely not right now) we need to leave a TODO/issue to revisit on error handling, since we will have bad streaming state kept in memory forever -> OOM